tensorboard/data/provider.py

```diff
@@ -120,12 +121,56 @@ def read_scalars(
     """
     pass
 
-  def list_tensors(self):
-    """Not yet specified."""
+  def list_tensors(self, experiment_id, plugin_name, run_tag_filter=None):
+    """List metadata about tensor time series.
+
+    Args:
+      experiment_id: ID of enclosing experiment.
+      plugin_name: String name of the TensorBoard plugin that created
+        the data to be queried. Required.
+      run_tag_filter: Optional `RunTagFilter` value. If omitted, all
+        runs and tags will be included.
+
+    The result will only contain keys for run-tag combinations that
+    actually exist, which may not include all entries in the
+    `run_tag_filter`.
+
+    Returns:
+      A nested map `d` such that `d[run][tag]` is a `TensorTimeSeries`
+      value.
+
+    Raises:
+      tensorboard.errors.PublicError: See `DataProvider` class docstring.
+    """
     pass
 
-  def read_tensors(self):
-    """Not yet specified."""
+  def read_tensors(
+      self, experiment_id, plugin_name, downsample=None, run_tag_filter=None
+  ):
+    """Read values from tensor time series.
+
+    Args:
+      experiment_id: ID of enclosing experiment.
+      plugin_name: String name of the TensorBoard plugin that created
```

Comment:
No action required at all but asking for a short discussion.

It feels weird that a data is tied to a plugin. I think we should call this `summary_name`. Of course, I think there should be a equivalent changes on the summary API. What do you think?

Reply:
Heh. Yes, agreed; it’s not clear whether “plugin name” is a better
ontology than something more like “data type” or “MIME type”, where
there is potentially a many-to-many relationship between producers
(plugins) and consumers (dashboards) of a given data type.

As you observe, there’s not really a good place to fit this into the
current summary APIs, so I opted to be internally consistent. I can
certainly imagine a world in which we intentionally coöpt the semantics
of `plugin_name` to mean something more like “MIME type”.

I suspect that `summary_name` could be a bit confusing; I’d be inclined
to say that the “name” of my summary is `loss` or `accuracy`, not
`scalars` or `images`.

Would be happy to discuss more, and I think that @davidsoergel also has
some thoughts here.

Reply:
FWIW William and I talked about this a while back as well.  I still think ultimately we want something more like data type or MIME type, but we don't really have a coherent plan for this yet, so the thought was that we'd stick to consistency with the existing pattern of using plugin names until we have a better strategy.